engine/collection/synchronization/engine.go

```diff
@@ -54,310 +67,259 @@ func New(
 		participants: participants.Filter(filter.Not(filter.HasNodeID(me.NodeID()))),
-		state:        state,
-		blocks:       blocks,
 		comp:         comp,
 		core:         core,
-		pollInterval: 8 * time.Second,
-		scanInterval: 2 * time.Second,
+		pollInterval: opt.PollInterval,
+		scanInterval: opt.ScanInterval,
+		state:        state,
+	}
+
+	err := e.setupResponseMessageHandler()
+	if err != nil {
+		return nil, fmt.Errorf("could not setup message handler")
 	}
 
 	chainID, err := state.Params().ChainID()
 	if err != nil {
 		return nil, fmt.Errorf("could not get chain ID: %w", err)
 	}
 
-	e.chainID = chainID
-
 	// register the engine with the network layer and store the conduit
-	conduit, err := net.Register(engine.ChannelSyncCluster(chainID), e)
+	con, err := net.Register(engine.ChannelSyncCluster(chainID), e)
 	if err != nil {
 		return nil, fmt.Errorf("could not register engine: %w", err)
 	}
-	e.conduit = conduit
+	e.con = con
+
+	e.requestHandler = NewRequestHandlerEngine(log, metrics, con, me, blocks, core, state)
 
 	return e, nil
 }
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started. For consensus engine, this is true once the underlying consensus
-// algorithm has started.
+// setupResponseMessageHandler initializes the inbound queues and the MessageHandler for UNTRUSTED responses.
+func (e *Engine) setupResponseMessageHandler() error {
+	syncResponseQueue, err := fifoqueue.NewFifoQueue(
+		fifoqueue.WithCapacity(defaultSyncResponseQueueCapacity))
+	if err != nil {
+		return fmt.Errorf("failed to create queue for sync responses: %w", err)
+	}
+
+	e.pendingSyncResponses = &engine.FifoMessageStore{
+		FifoQueue: syncResponseQueue,
+	}
+
+	blockResponseQueue, err := fifoqueue.NewFifoQueue(
+		fifoqueue.WithCapacity(defaultBlockResponseQueueCapacity))
+	if err != nil {
+		return fmt.Errorf("failed to create queue for block responses: %w", err)
+	}
+
+	e.pendingBlockResponses = &engine.FifoMessageStore{
+		FifoQueue: blockResponseQueue,
+	}
+
+	// define message queueing behaviour
+	e.responseMessageHandler = engine.NewMessageHandler(
+		e.log,
+		engine.NewNotifier(),
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*messages.SyncResponse)
+				if ok {
+					e.metrics.MessageReceived(metrics.EngineClusterSynchronization, metrics.MessageSyncResponse)
+				}
+				return ok
+			},
+			Store: e.pendingSyncResponses,
+		},
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*messages.ClusterBlockResponse)
+				if ok {
+					e.metrics.MessageReceived(metrics.EngineClusterSynchronization, metrics.MessageBlockResponse)
+				}
+				return ok
+			},
+			Store: e.pendingBlockResponses,
+		},
+	)
+
+	return nil
+}
+
+// Ready returns a ready channel that is closed once the engine has fully started.
 func (e *Engine) Ready() <-chan struct{} {
-	e.unit.Launch(e.checkLoop)
-	return e.unit.Ready()
+	e.lm.OnStart(func() {
```

Comment:
Since we are using `unit` it is safer to first invoke `unit.Ready`. This follows the lifecycle assumptions of the `unit`.